src/tovnode.ts

```diff
@@ -19,8 +19,8 @@ export function toVNode(node: Node, domApi?: DOMAPI): VNode {
   let text: string;
   if (api.isElement(node)) {
     const id = node.id ? "#" + node.id : "";
-    const cn = node.getAttribute("class");
-    const c = cn ? "." + cn.split(" ").join(".") : "";
+    const cn = node.getAttribute("class")?.trim();
+    const c = cn ? "." + cn.match(/[^\t\r\n\f ]+/g)!.join(".") : "";
```

Comment:
If I understand correctly the `trim` is to handle the only-whitespace case.

Would this work as well?

```diff
-    const cn = node.getAttribute("class")?.trim();
-    const c = cn ? "." + cn.match(/[^\t\r\n\f ]+/g)!.join(".") : "";
+    const classNames = node.getAttribute("class")?.match(/[^\t\r\n\f ]+/g);
+    const c = classNames ? "." + classNames.join(".") : "";
```

This would avoid the `trim` and the use of `!`.

Reply:
Good idea, will do that